org.lflang/src/org/lflang/generator/c/CPreambleGenerator.java

```diff
@@ -71,6 +77,12 @@ public static String generateDefineDirectives(
         CodeBuilder code = new CodeBuilder();
         code.pr("#define LOG_LEVEL " + logLevel);
         code.pr("#define TARGET_FILES_DIRECTORY " + addDoubleQuotes(srcGenPath.toString()));
+
+        //TODO: Define Global Support for 32-bit and Microsecond Granularity on Reactor-C
+        if(targetConfig.platform == Platform.ARDUINO) {
```

Comment:
I think all the Arduino-related code can be consolidated into a new class (e.g., called `org.lflang.generator.c.CArduinoSupport.java`) with various static methods (like `generateDefineStatements`). This would make it easy to expand the support infrastructure in the future. However, I don't think this is essential for merging this in.

Reply:
Can add this down the line when we have more features.